src/extensions/agent_modules/cloud_communication/eg_mqtt_broker_client/src/adu_enrollment_management.c

```diff
@@ -315,57 +223,44 @@ void OnMessage_enr_resp(
 
     // Parse the JSON payload using Parson
     root_value = json_parse_string(msg->payload);
-    root_object = json_value_get_object(root_value);
 
     if (root_value == NULL)
     {
         Log_Error("Failed to parse JSON");
         goto done;
     }
 
-    isEnrolled = json_object_dotget_boolean(root_object, "IsEnrolled");
-    duInstance = json_object_dotget_string(root_object, "ScopeId");
-
-    // Read 'resultcode', 'extendedresultcode' and 'resultdescription' from the response message properties.
-
-    // Validate received data
-    if (isEnrolled == -1) // json_object_dotget_boolean returns -1 on failure
+    if (!ADUC_JSON_GetBooleanField(root_value, "IsEnrolled", &isEnrolled))
     {
-        Log_Error("Failed to retrieve or validate isEnrolled from JSON payload");
+        Log_Error("Failed to get 'isEnrolled' from payload");
         goto done;
     }
 
-    // Store the enrollment response data in the state.
-    if (isEnrolled != 0)
+    duInstance = ADUC_JSON_GetStringFieldPtr(root_value, "ScopeId");
```

Comment:
what's "du" mean?  why isn't this scopeId, for example?

Reply:
device update I think. Changing the name to scopeId everywhere here except in the data store api.